src/core/externalstorage/qgshttpexternalstorage_p.h

```diff
@@ -55,43 +55,67 @@ class CORE_EXPORT QgsWebDAVExternalStorage : public QgsExternalStorage
 
 /**
  * \ingroup core
- * \brief Class for WebDAV stored content
+ * \brief External storage implementation using the protocol AWS S3.
+ *
+ * \since QGIS 3.28
+ */
+class CORE_EXPORT QgsAWSS3ExternalStorage : public QgsExternalStorage
+{
+  public:
+
+    QString type() const override;
+
+    QString displayName() const override;
+
+  protected:
+
+    QgsExternalStorageStoredContent *doStore( const QString &filePath, const QString &url, const QString &authcfg = QString() ) const override;
+
+    QgsExternalStorageFetchedContent *doFetch( const QString &url, const QString &authConfig = QString() ) const override;
+};
+
+/**
+ * \ingroup core
```

Comment:
```suggestion
```
If I'm not wrong, this is doxy private, you should probably remove \incore

Reply:
> If I'm not wrong, this is doxy private

You're right. This is coming from the original block, so I left it, but if you confirm I should remove it, I will.

Reply:
yes please, it will be ignored anyway.

Reply:
All right, done, in qgssimplecopyexternalstorage_p.h as well